tutorials/2d/custom_drawing_in_2d.rst

```diff
@@ -195,333 +209,850 @@ The same applies to the ``draw_rect`` method with ``filled = false``.
+    }
+
+It will look somewhat like this when run:
+
+.. image:: img/draw_godot_mouth_animation.webp
+
+Please note that ``_mouth_width`` is a user defined property like any other
+and it or any other used as a drawing argument can be animated using more
+standard and high level methods such as a :ref:`Tween<class_Tween>` or an
+:ref:`AnimationPlayer<class_AnimationPlayer>` Node. The only difference is
+that a ``queue_redraw()`` call is needed to apply those changes so they get
+shown on screen.
+
+.. _doc_draw_custom_example_2:
+
+Example 2: drawing a dynamic line
+---------------------------------
+
+The previous example was useful to learn how to draw and modify nodes with
+custom shapes and animations. This could have some advantages, such as using
+exact coordinates and vectors for drawing, rather than bitmaps -which means
+they will scale well when transformed on screen. In some cases, similar results
+could be achieved composing higher level functionality with nodes such as
+:ref:`sprites<class_Sprite2D>` or
+:ref:`AnimatedSprites<class_AnimatedSprite2D>` loading SVG resources (which are
+also images defined with vectors) and the
+:ref:`AnimationPlayer<class_AnimationPlayer>` node.
+
+In other cases that will not be possible because we will not know what the
+resulting graphical representation will be before running the code. Here we
+will see how to draw a dynamic line whose coordinates are not known beforehand,
+and are affected by the user's input.
+
+Drawing a straight line between 2 points
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's assume we want to draw a straight line between 2 points, the first one
+will be fixed on the upper left corner ``(0, 0)`` and the second will be defined
+by the cursor position on screen.
+
+We could draw a dynamic line between those 2 points like this:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends Node2D
+
+    var point1 : Vector2 = Vector2(0, 0)
+    var width : int = 10
+    var color : Color = Color.GREEN
+
+    var _point2 : Vector2
 
-        // We only wrap angles when both of them are bigger than 360.
-        if (_angleFrom > 360 && _angleTo > 360)
+    func _process(_delta):
+        var mouse_position = get_viewport().get_mouse_position()
+        if mouse_position != _point2:
+            _point2 = mouse_position
+            queue_redraw()
+
+    func _draw():
+        draw_line(point1, _point2, color, width)
+
+ .. code-tab:: csharp
+
+    using Godot;
+    using System;
+
+    public partial class MyNode2DLine : Node2D
+    {
+        public Vector2 point1 = new Vector2(0f, 0f);
+        public int width = 10;
+        public Color color = Colors.Green;
+
+        private Vector2 point2;
+
+        public override void _Process(double delta)
         {
-            _angleFrom = Wrap(_angleFrom, 0, 360);
-            _angleTo = Wrap(_angleTo, 0, 360);
+            Vector2 mouse_position = GetViewport().GetMousePosition();
+            if (mouse_position != point2)
```

Comment:
Vector2 members are float, so using the `==` or `!=` operators may not work as expected due to floating-point precision errors. You may want to use `IsEqualApprox` instead.

```suggestion
            if (mousePosition != _point2)
```